src/proof/types.rs

```diff
@@ -1,56 +1,91 @@
 use super::*;
-use rlp::{RlpStream};
-use ethereum_types::{Bloom, H160, H256, H64, U256};
+use ethereum_types::{Address, Bloom, H256, H64, U256};
+use rlp::RlpStream;
+use rlp_derive::{RlpDecodable, RlpEncodable};
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, RlpEncodable, RlpDecodable)]
 pub struct BlockHeader {
-	pub parent_hash: H256,
-	pub uncles_hash: H256,
-	pub author: H160,
-	pub state_root: H256,
-	pub transactions_root: H256,
-	pub receipts_root: H256,
-	pub log_bloom: Bloom,
-	pub difficulty: U256,
-	pub number: u64,
-	pub gas_limit: U256,
-	pub gas_used: U256,
-	pub timestamp: u64,
-	pub extra_data: Vec<u8>,
-	pub mix_hash: H256,
-	pub nonce: H64,
+    pub parent_hash: H256,
+    pub uncles_hash: H256,
+    pub author: Address,
+    pub state_root: H256,
+    pub transactions_root: H256,
+    pub receipts_root: H256,
+    pub log_bloom: Bloom,
+    pub difficulty: U256,
+    pub number: U256,
+    pub gas_limit: u64,
+    pub gas_used: u64,
+    pub timestamp: u64,
+    pub extra_data: String,
+    pub mix_hash: H256,
+    pub nonce: H64,
+}
+
+#[derive(Debug, Clone, RlpEncodable, RlpDecodable)]
+pub struct BlockHeaderSeal {
+    pub parent_hash: H256,
+    pub uncles_hash: H256,
+    pub author: Address,
+    pub state_root: H256,
+    pub transactions_root: H256,
+    pub receipts_root: H256,
+    pub log_bloom: Bloom,
+    pub difficulty: U256,
+    pub number: U256,
+    pub gas_limit: u64,
+    pub gas_used: u64,
+    pub timestamp: u64,
+    pub extra_data: String,
+}
 
-	pub hash: Option<H256>,
-	pub partial_hash: Option<H256>,
+impl From<BlockHeader> for BlockHeaderSeal {
+    fn from(b: BlockHeader) -> Self {
+        Self {
+            parent_hash: b.parent_hash,
+            uncles_hash: b.uncles_hash,
+            author: b.author,
+            state_root: b.state_root,
+            transactions_root: b.transactions_root,
+            receipts_root: b.receipts_root,
+            log_bloom: b.log_bloom,
+            difficulty: b.difficulty,
+            number: b.number,
+            gas_limit: b.gas_limit,
+            gas_used: b.gas_used,
+            timestamp: b.timestamp,
+            extra_data: b.extra_data,
+        }
+    }
 }
 
 impl BlockHeader {
-	pub fn extra_data(&self) -> H256 {
-		let mut data = [0u8; 32];
-		data.copy_from_slice(self.extra_data.as_slice());
-		H256(data.into())
-	}
+    pub fn extra_data(&self) -> H256 {
```

Comment:
spacing should be tabs of 4 spaces here, looks like its 2.

Reply:
I will configure it using `rustfmt.toml` file.